go/sentry/sentry.go

```diff
@@ -82,6 +87,25 @@ func (b *backend) GetUpstreamTLSPubKeys(ctx context.Context) ([]signature.Public
 	return b.upstreamTLSPubKeys, nil
 }
 
+func (b *backend) UpdatePolicies(ctx context.Context, p grpcPolApi.ServicePolicies) error {
+	b.Lock()
+	defer b.Unlock()
+
+	b.grpcPolicyCheckers[p.Service] = policy.NewDynamicRuntimePolicyChecker(p.Service, nil)
+	for namespace, policy := range p.AccessPolicies {
+		b.grpcPolicyCheckers[p.Service].SetAccessPolicy(policy, namespace)
+	}
+
+	return nil
+}
+
+func (b *backend) GetPolicyCheckers(ctx context.Context) (map[cmnGrpc.ServiceName]*policy.DynamicRuntimePolicyChecker, error) {
+	b.RLock()
+	defer b.RUnlock()
+
+	return b.grpcPolicyCheckers, nil
```

Comment:
This returns a reference to the map so the above lock doesn't protect it. Making a copy of the whole map seems unnecessary, maybe just change the method to take a service name argument and only return a single policy checker (the `*policy.DynamicRuntimePolicyChecker` uses a mutex internally so sharing that should be fine)?